core/storage/wal.rs

```diff
@@ -1696,7 +1696,12 @@ impl WalFile {
             }
         }
 
-        self.last_checksum = self.get_shared().last_checksum;
+        let (header, cksm) = {
```

Comment:
should these steps happen before truncation? what if `restart_wal_header` succeeds and truncation fails?